vm_supervisor/network/ndp_proxy.py

```diff
@@ -0,0 +1,56 @@
+"""
+Neighbourhood Discovery Proxy (NDP) functionalities.
+
+Some cloud providers do not route the whole advertised IPv6 address range to servers, but instead
+only route one address. They will issue NDP requests to the network to determine if the other
+addresses in the range are used. This means that our server (be it the hypervisor or the VMs)
+has to answer to these requests to make the VMs routable.
+
+To achieve this, we use ndppd. Each time an update is required, we overwrite /etc/ndppd.conf
+and restart the service.
+"""
+import logging
+from dataclasses import dataclass
+from ipaddress import IPv6Network
+from pathlib import Path
+from subprocess import run
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class NdpRule:
+    address_range: IPv6Network
+
+
+class NdpProxy:
+    def __init__(self, host_network_interface: str):
+        self.host_network_interface = host_network_interface
+        self.interface_address_range_mapping = {}
```

Comment:
What is the goal of this (untyped) dict ?

Reply:
This is a cache to avoid having to read the `/etc/ndppd.conf` file. With this, when updating the config, we can just overwrite the file entirely. And when removing a mapping, we can just pop the corresponding entry from this dictionary and then regenerate the file.

Easier (and probably faster) than parsing the file.